### PR TITLE
*:fix issue 428

### PR DESCRIPTION
**Pull request by nolouch**
https://github.com/pingcap/tidb/issues/428

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

File: plan/validator.go [hunk 1]
```diff
@@ -269,13 +269,23 @@ func (v *validator) checkFieldList(x *ast.FieldList) {
...
 	for _, colDef := range stmt.Cols {
 		tp := colDef.Tp
 		if tp.Tp == mysql.TypeString &&
 			tp.Flen != types.UnspecifiedLength && tp.Flen > 255 {
 			v.err = errors.Errorf("Column length too big for column '%s' (max = 255); use BLOB or TEXT instead", colDef.Name.Name.O)
 			return
 		}
+		for _, op := range colDef.Options {
+			if op.Tp == ast.ColumnOptionPrimaryKey {
+				if hasPrimaryKey {
```
Comment (zimulala) [inline plan/validator.go]: Can you reduce the deep nesting?

**Comment by zimulala** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**